src/core/unit-dependency-atom.h

```diff
@@ -75,18 +75,21 @@ typedef enum UnitDependencyAtom {
         /* Trigger a dependency on unsuccessful service exit. */
         UNIT_ATOM_ON_FAILURE                          = UINT64_C(1) << 24,
 
+        /* Pass exit status to a triggered OnFailure=/OnSuccess= dependency */
+        UNIT_ATOM_PROPAGATE_EXIT_STATUS               = UINT64_C(1) << 25,
```

Comment:
maybe call this UNIT_ATOM_PROPAGATE_EXIT_STATUS_FROM?

i.e. the direction matters: the unit that has the dep is the one that the exit status shall be propagated to, and the unit the dep points to is where it should be propagated from.